modules/light-clients/06-solomachine/client_state.go

```diff
@@ -128,15 +128,21 @@ func (cs *ClientState) VerifyMembership(
 		return errorsmod.Wrapf(ibcerrors.ErrInvalidType, "expected %T, got %T", commitmenttypes.MerklePath{}, path)
 	}
 
-	if merklePath.Empty() {
-		return errorsmod.Wrap(commitmenttypes.ErrInvalidProof, "path is empty")
+	if len(merklePath.GetKeyPath()) != 2 {
+		return errorsmod.Wrapf(host.ErrInvalidPath, "path must be of length 2: %s", merklePath.GetKeyPath())
+	}
+
+	// in a multistore context: index 0 is the key for the IBC store in the multistore, index 1 is the key in the IBC store
+	key, err := merklePath.GetKey(1)
```

Comment:
`GetKey` seems to do some unescaping on the string also, is this okay here?

from `GetKey`:

```
key, err := url.PathUnescape(mp.KeyPath[i])
```

Reply:
Maybe we should just use `merklePath.KeyPath[1]`?

Reply:
wouldn't this be fine considering escaping calls have been removed? It seems like it would just be best to tweak `GetKey`.

Reply:
That's a good observation @damiannolan! 

I think it is still fine, since it is doing unescaping. And also according to ICS 24 `%` is not a valid character in identifiers and paths, so it should not happen that a path contains an escaped character and then `GetKey` would return it unescaped.

Reply:
What do you guys prefer to do?
-

Reply:
Happy with opt 2 or 3 I think!

Reply:
if we're having a vote I'd go towards 3.

Reply:
Ok, I will go for 2 and implement 3 in the next PR.

Reply:
Done. I will merge when tests pass.